python/python/glide/async_commands/server_modules/json.py

```diff
@@ -320,6 +324,103 @@ async def forget(
+        >>> await json.strappend(client, "doc", '"foo"', "nested.a")
+            11  # The length of the string value after appending "foo" to the string at path 'nested.array' in the key stored at `doc`.
+        >>> jsonpy.loads(await json.get(client, jsonpy.dumps("doc"), "$"))
+            [{"a":"foobaz", "nested": {"a": "hellobazfoo"}, "nested2": {"a": 31}}] # The updated JSON value in the key stored at `doc`.
+    """
+
+    return cast(
+        TJsonResponse[int],
+        await client.custom_command(
+            ["JSON.STRAPPEND", key] + ([path, value] if path else [value])
+        ),
+    )
+
+
+async def strlen(
+    client: TGlideClient,
+    key: TEncodable,
+    path: Optional[TEncodable] = None,
+) -> TJsonResponse[Optional[int]]:
+    """
+    Returns the length of the JSON string value stored at the specified `path` within the JSON document stored at `key`.
+
+    Args:
+        client (TGlideClient): The client to execute the command.
+        key (TEncodable): The key of the JSON document.
+        path (Optional[TEncodable]): The path within the JSON document. Default to None.
+
+    Returns:
+        TJsonResponse[Optional[int]]:
+            For JSONPath (`path` starts with `$`):
+                Returns a list of integer replies for every possible path, indicating the length of the JSON string value,
+                or None for JSON values matching the path that are not string.
+            For legacy path (`path` doesn't start with `$`):
+                Returns the length of the JSON value at `path` or None if `key` doesn't exist.
+                If multiple paths match, the length of the first mached string is returned.
+                If the JSON value at `path` is not a string of if `path` doesn't exist, an error is raised.
+            If `key` doesn't exist, None is returned.
+        For more information about the returned type, see `TJsonResponse`.
+
+    Examples:
+        >>> from glide import json
+        >>> import jsonpy
+        >>> await json.set(client, "doc", "$", jsonpy.dumps({"a":"foo", "nested": {"a": "hello"}, "nested2": {"a": 31}}))
+            'OK'
+        >>> await json.strlen(client, "doc", "$..a")
+            [3, 5, None]  # The length of the string values at path '$..a' in the key stored at `doc`.
+        >>> await json.strlen(client, "doc", "nested.a")
+            5  # The length of the JSON value at path 'nested.a' in the key stored at `doc`.
+        >>> await json.strlen(client, "doc", "$")
+            [None]  # Returns an array with None since the value at root path does in the JSON document stored at `doc` is not a string.
+        >>> await json.strlen(client, "non_existing_key", ".")
+            None  # `key` doesn't exist.
+    """
+
+    return cast(
+        TJsonResponse[Optional[int]],
+        await client.custom_command(
+            ["JSON.STRLEN", key, path] if path else ["JSON.STRLEN", key]
```

Comment:
nit: personally I don't like these shortcut forms because they're specific to python